addons/account_edi/models/account_move.py

```diff
@@ -160,8 +169,14 @@ def button_cancel(self):
         # Set the electronic document to be canceled and cancel immediately for synchronous formats.
         res = super().button_cancel()
 
-        self.edi_document_ids.filtered(lambda doc: doc.attachment_id).write({'state': 'to_cancel', 'error': False})
-        self.edi_document_ids.filtered(lambda doc: not doc.attachment_id).write({'state': 'cancelled', 'error': False})
+        values = {'state': 'to_cancel', 'error': False}
+        if 'blocked_level' in self.env['account.edi.document']._fields:
+            values['blocked_level'] = False
+        self.edi_document_ids.filtered(lambda doc: doc.attachment_id).write(values)
+        values = {'state': 'cancelled', 'error': False}
+        if 'blocked_level' in self.env['account.edi.document']._fields:
+            values['blocked_level'] = False
```

Comment:
shouldn't be simpler to manage `blocked_level` all the time and drop it into an override of `create`/`write` of `account.edi.document` ?